pkg/cluster/context.go

```diff
@@ -63,9 +63,17 @@ func NewContext(name string) (ctx *Context, err error) {
 			name, validNameRE.String(),
 		)
 	}
+	return newContextNoValidation(name), nil
```

Comment:
actually New() probably should not validate or error and there should be a separate validation call when creating things, I'll do this in some follow-up

Reply:
added a TODO